packages/esm-patient-chart-app/src/mark-patient-deceased/deceased-form.workspace.tsx

```diff
@@ -0,0 +1,240 @@
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const filteredCausesOfDeath = useMemo(() => {
+    if (!searchTerm) {
+      return causesOfDeath;
+    }
+
+    return searchTerm
+      ? fuzzy
+          .filter(searchTerm, causesOfDeath, {
+            extract: (causeOfDeathConcept) => causeOfDeathConcept.display,
+          })
+          .sort((r1, r2) => r1.score - r2.score)
+          .map((result) => result.original)
+      : causesOfDeath;
+  }, [searchTerm, causesOfDeath]);
+
+  const schema = z.object({
+    causeOfDeath: z.string(),
+    deathDate: z.date().nullable(),
+  });
+
+  type MarkPatientDeceasedFormSchema = z.infer<typeof schema>;
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    setError,
+  } = useForm<MarkPatientDeceasedFormSchema>({
+    mode: 'onSubmit',
+    resolver: zodResolver(schema),
+    defaultValues: {
+      causeOfDeath: '',
+      deathDate: isDead
+        ? new Date(dayjs(deathDate).year(), dayjs(deathDate).month(), dayjs(deathDate).date() - 1)
+        : new Date(),
+    },
+  });
+
+  const onSubmit: SubmitHandler<MarkPatientDeceasedFormSchema> = useCallback(
+    (data) => {
+      setIsSubmitting(true);
+      const { causeOfDeath, deathDate } = data;
+
+      if (!causeOfDeath || !deathDate) {
+        if (!deathDate) {
+          setError('deathDate', {
+            type: 'manual',
+            message: t('deathDateRequired', 'Please select the date of death'),
+          });
+        }
+
+        if (!causeOfDeath) {
+          setError('causeOfDeath', {
+            type: 'manual',
+            message: t('causeOfDeathIsRequired', 'Please select the cause of death'),
+          });
+        }
+        setIsSubmitting(false);
+        return;
+      }
+
+      markPatientDeceased(deathDate, patientUuid, causeOfDeath)
+        .then(() => {
+          closeWorkspace();
+          window.location.reload();
+        })
+        .catch((error) => {
+          showSnackbar({
+            kind: 'error',
+            isLowContrast: false,
+            subtitle: error?.message,
+            title: t('errorMarkingPatientDeceased', 'Error marking patient deceased'),
+          });
+        })
+        .finally(() => {
+          setIsSubmitting(false);
+        });
+    },
+    [closeWorkspace, patientUuid, setError, t],
+  );
+
+  const onError = (errors) => console.error(errors);
+
+  return (
+    <Form className={styles.form} onSubmit={handleSubmit(onSubmit, onError)}>
+      <div>
+        {isTablet && (
+          <Row className={styles.headerGridRow}>
+            <ExtensionSlot className={styles.dataGridRow} name="visit-form-header-slot" state={memoizedPatientUuid} />
+          </Row>
+        )}
+        <div className={styles.container}>
+          <span className={styles.warningContainer}>
+            <WarningFilled aria-label={t('warning', 'Warning')} className={styles.warningIcon} size={20} />
```

Comment:
@ibacher should we switch to the new icons in the framework across the board?

Reply:
As much as possible! I haven't created a component for every SVG yet and there are some gaps (warning--filled is one of the icons not included in CarbonMRS and I'm not entirely certain we can just re-use Carbon icons, but it's something I'll look into).